core/src/subgraph/runner.rs

```diff
@@ -468,6 +468,7 @@ where
         } = block_state;
 
         let first_error = deterministic_errors.first().cloned();
+        let deterministic_errors = Arc::new(deterministic_errors);
```

Comment:
You shouldn't put this into an `Arc`; that way, you don't have to call `clone().to_vec()` below and can just pass `deterministic_errors` into `transact_block_operations`

Reply:
got it, thanks for pointing it out, did so since previously non_fatal_errors were being written in the same function and deterministic_errors get moved, now we do that in `transact_block_operations` so its no longer required to be Arc